control/controlclient/map.go

```diff
@@ -828,6 +829,23 @@ func (ms *mapSession) sortedPeers() []tailcfg.NodeView {
 func (ms *mapSession) netmap() *netmap.NetworkMap {
 	peerViews := ms.sortedPeers()
 
+	var displayMessages map[tailcfg.DisplayMessageID]tailcfg.DisplayMessage
+
+	if len(ms.lastHealth) > 0 {
+		// As they all resolve to the same ID, we can only pass on one message
+		// from ControlHealth. In practice we generally send 0 or 1, but
+		// if there are multiple, the last one wins.
```

Comment:
This seems surprising. We're nerfing that MapResponse.Health field with this, no? And the docs on that field aren't updated to say that it only supports one now?

But why can't this support more than one? can't we tack on a numeric index or truncated hash of the text or something? Maybe as a fallback for being unable to parse out a code from the string, in some new documented way? Like maybe at a certain cap ver, we tell the control plane we'll parse out the `[code]` from `"foo bar [code]`" or something?

But seems like `control-health-1`, `control-health-2`, etc is easy enough?

Reply:
As of dfbb75d0415c68d2de858b15089dcc21e727e886, all control health messages now show up in CurrentState().Warnings where the key is `"control-health-"+sha256(msg)`